docs/reference/projectyml.rst

```diff
@@ -50,66 +50,122 @@ The code snippet below provides an example of a complete ``project.yml`` file:
+.. code-block:: yaml
 
+    pip:
+        index-url: https://devpi.example.com/stable/
+        extra-index-url: []
+        pre: false
+        use-system-config: true
 
-Alternatively, a pip config file can be used at a custom location.
-The ``index-url`` can be specified in this file as explained in the previous section.
-To make this work, the ``PIP_CONFIG_FILE`` environment variable needs to be set to the path of the newly created file (See: :ref:`env_vars`).
-For more information see the `Pip documentation <https://pip.pypa.io/en/stable/topics/configuration/>`_.
+.. note::
+    The options defined in the ``project.yml`` pip section will always take precedence over the corresponding pip options, even
+    when ``use-system-config`` is set to true (other pip-related environment variables are not overridden).
 
-Specify the index-urls in the project.yml file
-""""""""""""""""""""""""""""""""""""""""""""""
+    For example, in the production scenario above, if the following
+    pip environment variables were set by mistake on the server running the compiler: ``PIP_INDEX_URL=https://devpi.example.com/dev/``
+    and ``PIP_PRE=true``, the config used in the end would still be the one defined in the project.yml, namely
+    ``index-url=https://devpi.example.com/stable/`` and ``pre=false``.
 
-Another option is to use the  ``index_urls`` option in the ``pip`` section of the ``project.yml`` file:
+An alternative approach would be to configure all pip-related options through the system config. For example:
 
 .. code-block:: yaml
 
     pip:
-      use_config_file: False
-      index_urls:
-        - <url of a python package repository>
+        index-url:
+        extra-index-url: []
+        pre:
+        use-system-config: true
+
+And set the following env variables:
 
+.. code-block:: bash
+
+    export PIP_INDEX_URL=https://devpi.example.com/stable/
+    export PIP_PRE=false
+
+In this scenario, pip options defined in env variables would be used over the system's pip config (if any).
 
 .. note::
-    The pip config file can also be used in combination with ``index-urls`` specified in the ``pip`` section of the ``project.yml`` file:
 
-    * If the pip config is used (by setting ``use_config_file`` to ``true``), the ``index-url`` specified in the pip config file will take precedence and the ``index-urls`` specified in the ``pip`` section of the ``project.yml`` file will be used as ``extra-index-urls`` when installing with pip.
-    * If the pip config is not used (by setting ``use_config_file`` to ``False``), then the first ``index_url`` specified in the project.yml will be used as an ``index_url`` and all the following ones will be used as ``extra-index-urls`` when installing with pip.
+    Using netrc is the recommended way to set up authentication towards the index. See
+    this :ref:`section<setting_up_pip_index_authentication>` for more information.
+
+.. _migrate_to_project_wide_pip_config:
+
+Migrate to project-wide pip config
```

Comment:
Should we also document changes relative to `inmanta-2023.4` (OSS), which already has `use-system-config` but in another location (I think) and somewhat bugged (or at least very unintuitive)?